libpod/container_internal_linux.go

```diff
@@ -752,8 +752,19 @@ func (c *Container) makeBindMounts() error {
 
 // generateResolvConf generates a containers resolv.conf
 func (c *Container) generateResolvConf() (string, error) {
+	resolvConf := "/etc/resolv.conf"
+	if c.config.NetMode.IsUserDefined() {
```

Comment:
No need for this.

Check c.config.Spec.Linux.Namespaces - we can find the network namespace in there, and check if it's set to a path (if it's set to "", we make our own network namespace, so no need for this handling). If it's set to a path, take filepath.Base on it and move on with the logic here

Reply:
how would it work if the path is set to something like `/proc/PID/ns/net`?

Reply:
@mheon Not sure what you are saying?  Should I use
c.config.Spec.Linux.Namespaces["net].IsUserDefined() instead of c.config.NetMode.IsUserDefined()?

@giuseppe It would look for /etc/netns/net/resolv.conf if it is existed and fall back to /etc/resolv.conf.

Reply:
Also why is what I have bad?

Reply:
I'm trying to avoid adding stuff to the DB when we can avoid it - we've mostly avoided scalability issues from JSON, but no reason to invite problems.

@giuseppe Would it be enough to check for /proc/ paths and ignore those?

Reply:
@rhatdan Instead of IsUserDefined(), c.config.Spec.Linux.Namespaces will give you a https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L170-L177

The first field will always be `"network"` (network namespace), and the second can be `""` (in which case we do nothing) or a path (in which case we grab filepath.Base on it and continue)

Reply:
That is what we are using now.